### ISSUE TITLE
Conditions do not work with ES6 arrow functions

### ISSUE THREAD

**Issue by Zelgadis87** [anywhichway/rule-reactor#19]
When using ES6 arrow functions as a rule condition, a TypeError is thrown at runtime stating that "Condition function missing a return statereactornt in rule X" (I assume statereactornt is a typo, and is actually meant to be statement).  
By removing the check for a return statement, the rule works perfectly (assuming the function is an arrow function, of course).

Example rule:
```
reactor.createRule( 'I have a message update', 0, { u: reactor.domain.Update }, [
  ( u ) => u.message
], ( u ) => {
  console.log( 'Received message update' );
  reactor.retract( u );

  [...]
} );
```

**Comment by anywhichway** [comment]
@Zelgadis87 you are correct, there is a typo in the error message. Obviously, if you are running a fork changing the source is fine. However, if would be safer to simply put a return in your arrow function, which is perfectly legal, although verbose. Antecedent conditions really ought to return a boolean, which is why we check for some type of return statement. However, we are trying to keep the rule "compiler" and run-time checking as light as possible. I will leave this issue open for comment by others. With sufficient input we will consider a change to the default behavior. Thanks for trying out rule-reactor.

**Comment by yellow1912** [comment]
I run into the same problem if I were to use a function with bind. For example:
```

let fn = function () {
                                return this.fn.apply(this, this.args);
                            };

                            fn = fn.bind({fn: this.__proto__[condition], args: conditions[i]});

                            rules.push(fn);
```

With bind, the returned result is not a string that you can use indexOf to check for return. I honestly think you should not check for this, you can check for error in run time but not here.

**Comment by anywhichway** [comment]
The behavior of rule-reactor will be ill-defined with arrow function s and bind. This is because rule-reactor uses a pseudo-compiler to analyze rule structure that looks for the "function" key word. There is no reliable way for us to work around this. This is marked as a won't fix.

**Comment by yellow1912** [comment]
But is it really necessary to check this at compile time? Isn't it the developer's responsibility to ensure the defined function return the expected result? 

For example, your code checks for the presence of the 'return' keyword, but what if I return a non boolean value? In that case you have no way to check it. So what's the reason of the check if you cannot prevent all errors?

**Comment by anywhichway** [comment]
I guess is really depends on ones perspective regarding compilation. We will look at just changing this to a warning given the increased prevalence of arrow function use in the world. Initially theses will not be suppressible, so you will get noise in the console window or stdio.

**Comment by anywhichway** [comment]
@yellow1912 and @Zelgadis87 This has been fixed in v0.1.11. It will be pushed to NPM once one of you have verified.

**Comment by anywhichway** [comment]
Pushed to npmjs
